File: libs/gomodels/channel_role_user.go
package model

type ChannelRoleUser struct {
	ID     string `gorm:"column:id;primary_key;type:uuid;default:uuid_generate_v4()" json:"id"`
	UserID string `gorm:"column:userId;type:uuid;" json:"userId"`
	RoleID string `gorm:"column:roleId;type:uuid;" json:"-"`

	Role *ChannelRole `gorm:"foreignKey:RoleID" json:"-"`
	User *Users       `gorm:"foreignKey:UserID" json:"-"`
}

func (ChannelRoleUser) TableName() string {
	return "channels_roles_users"
}

File: libs/typeorm/src/entities/ChannelRoleUser.ts
/* eslint-disable import/no-cycle */
import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';

import { ChannelRole } from './ChannelRole';
import { User } from './User';

@Entity('channels_roles_users')
export class ChannelRoleUser {
    @PrimaryGeneratedColumn('uuid')
    id: number;

    @Column()
    userId: number;

    @Column()
    roleId: number;

    @ManyToOne(() => ChannelRole, _ => _.users, { onDelete: 'CASCADE' })
    @JoinColumn({ name: 'roleId' })
    role?: ChannelRole;

    @ManyToOne(() => User, _ => _.channelRoleUsers)
    @JoinColumn({ name: 'userId' })
    user?: User;
}